src/components/Routes.jsx

```diff
@@ -0,0 +1,87 @@
+import { Route, Switch } from 'react-router-dom';
+import UsersWithContext from './ContextHOCs/UsersWithContext';
+import Tasks from '../pages/Tasks';
+import UsersTasksWithContext from './ContextHOCs/UsersTasksWithContext';
+import UsersTracks from '../pages/UsersTracks';
+import UsersProgress from '../pages/UsersProgress';
+import PageNotFound from '../pages/PageNotFound';
+import settingsIcon from '../icons/settings.svg';
+import usersIcon from '../icons/user.svg';
+import tasksIcon from '../icons/tasks.svg';
+import MenuItem from './Aside/Menu/MenuItem/MenuItem';
+import SettingsWithContext from './ContextHOCs/SettingsWithContext';
+
+const paths = [
+  { isExact: true, path: '/main/users', component: UsersWithContext },
+  { isExact: true, path: '/main/tasks', component: Tasks },
+  { isExact: true, path: '/main/users/:userId/tasks', component: UsersTasksWithContext },
+  { isExact: true, path: '/main/users/:userId/tasks/:taskId/track', component: UsersTracks },
+  { isExact: true, path: '/main/users/:userId/progress', component: UsersProgress },
+  { isExact: true, path: '/main/settings', component: SettingsWithContext },
+  { isExact: false, path: '', component: PageNotFound },
+];
+
+const adminAndMentorPermissions = [0, 1, 2, 3, 4, 5, 6];
+const userPermissions = [2, 3, 5, 6];
```

Comment:
As I said it's bad approach, please change it, we should avoid using some kind of magic numbers, especially strange array of numbers.

Reply:
According to your advice, it was perfectly reworked! Just look on it!
```suggestion
const paths = [
  { isExact: true, path: '/main/users', permissions: [ADMIN, MENTOR], component: UsersWithContext },
  { isExact: true, path: '/main/tasks', permissions: [ADMIN, MENTOR], component: Tasks },
  {
    isExact: true,
    path: '/main/users/:userId/tasks',
    permissions: [ADMIN, MENTOR, USER],
    component: UsersTasksWithContext,
  },
  {
    isExact: true,
    path: '/main/users/:userId/tasks/:taskId/track',
    permissions: [ADMIN, MENTOR, USER],
    component: UsersTracks,
  },
  { isExact: true, path: '/main/users/:userId/progress', permissions: [ADMIN, MENTOR], component: UsersProgress },
  { isExact: true, path: '/main/settings', permissions: [ADMIN, MENTOR, USER], component: SettingsWithContext },
  { isExact: false, path: '', permissions: [ADMIN, MENTOR, USER], component: PageNotFound },
];
```